geopandas/io/sql.py

```diff
@@ -149,37 +151,53 @@ def _get_srid_from_crs(gdf):
     # Use geoalchemy2 default for srid
     # Note: undefined srid in PostGIS is 0
     srid = -1
+    warning_msg = (
+        "Could not parse CRS from the GeoDataFrame. "
+        + "Inserting data without defined CRS.",
+    )
     if gdf.crs is not None:
         try:
             srid = gdf.crs.to_epsg(min_confidence=25)
             if srid is None:
                 srid = -1
+                warnings.warn(warning_msg, UserWarning, stacklevel=2)
         except Exception:
             warnings.warn(
-                "Warning: Could not parse CRS from the GeoDataFrame.",
-                "Inserting data without defined CRS.",
-                UserWarning,
-                stacklevel=2,
+                warning_msg, UserWarning, stacklevel=2,
             )
     return srid
 
 
 def _convert_linearring_to_linestring(gdf, geom_name):
     from shapely.geometry import LineString
 
+    # Todo: Use Pygeos function once it's implemented:
+    #  https://github.com/pygeos/pygeos/issues/76
+
     mask = gdf.geom_type == "LinearRing"
     gdf.loc[mask, geom_name] = gdf.loc[mask, geom_name].apply(
         lambda geom: LineString(geom)
     )
     return gdf
 
 
-def _convert_to_wkb(gdf, geom_name):
-    """Convert geometries to wkb. """
-    from geopandas.array import from_shapely, to_wkb
+def _convert_to_ewkb(gdf, geom_name, srid):
+    """Convert geometries to ewkb. """
+    if compat.USE_PYGEOS:
+        from pygeos import from_shapely, set_srid, to_wkb
+
+        geoms = to_wkb(
+            set_srid(from_shapely(gdf[geom_name].values), srid=srid),
```

Comment:
```suggestion
            set_srid(gdf[geom_name].values.data, srid=srid),
```

(when using pygeos, the values in the column are already pygeos geometries, so no `from_shapely` is needed)

Reply:
Ah, still need some practice working with the pygeos version of gdp. Fixed this as well 👍